### PR TITLE
Fix bug: Binary data is not properly encoded.

### PR DESCRIPTION
**Pull request by shiweiz**
Binary data should be encoded using base64.

Previous conversions see PR #908.

### CODE REVIEW

**Comment by shiweiz** [comment]
@jhendrixMSFT

File: storage/entity.go [hunk 1]
```diff
@@ -246,7 +247,7 @@ func (e *Entity) MarshalJSON() ([]byte, error) {
 		switch t := v.(type) {
 		case []byte:
 			completeMap[typeKey] = OdataBinary
-			completeMap[k] = string(t)
+			completeMap[k] = t
```
Comment (jhendrixMSFT) [inline storage/entity.go]: This doesn't need to be base-64 encoded?

Comment (jhendrixMSFT) [reply storage/entity.go]: I suppose it already is.

Comment (shiweiz) [reply storage/entity.go]: t is of type []byte, and golang automatically converts []byte to base64 string when calling json.Marshal().

**Pull request merged**